frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts

```diff
@@ -127,6 +172,172 @@ export class ExperimentDesignStepperService {
+
+    const levelOrder = {};
+    existingPartitions.forEach((partition) => {
+      partition.factors.forEach((factor) => {
+        factor.levels.forEach((level) => {
+          levelOrder[level.id] = factor.order;
+        });
+      });
+    });
+
+    const tableData = existingConditions.map((factorialCondition) => {
+      const conditionAlias = existingConditionAliases.find(
+        (conditionAlias) => conditionAlias?.parentCondition.id === factorialCondition.id
+      );
+
+      const aliasname = conditionAlias ? conditionAlias.aliasName : '';
+      const existingConditionAliasId = conditionAlias?.id;
+
+      factorialCondition.levelCombinationElements.sort((a, b) =>
+        levelOrder[a.level?.id] > levelOrder[b.level?.id]
+          ? 1
+          : levelOrder[b.level?.id] > levelOrder[a.level?.id]
+          ? -1
+          : 0
+      );
+
+      const tableRow: FactorialConditionTableRowData = {
+        id: factorialCondition.id,
+        conditionAliasId: existingConditionAliasId,
+        levels: factorialCondition.levelCombinationElements.map((levelElement) => {
+          return {
+            id: levelElement.level.id,
+            name: levelElement.level.name,
+          };
+        }),
+
+        alias: aliasname,
+        weight: factorialCondition.assignmentWeight.toString(),
+        include: factorialCondition.assignmentWeight > 0,
+      };
+      return tableRow;
+    });
+    return tableData;
+  }
+
+  createFactorialConditionRequestObject() {
+    const tableData = this.getFactorialConditionTableData();
+    const factorialConditionsRequestObject: FactorialConditionRequestObject[] = [];
+    let conditionIndex = 1;
+    tableData.forEach((factorialConditionTableRow) => {
+      factorialConditionsRequestObject.push({
+        id: factorialConditionTableRow.id,
+        name: 'condition ' + conditionIndex, // what should this be?
+        conditionCode: 'condition ' + conditionIndex, // what should this be?
+        assignmentWeight: parseFloat(factorialConditionTableRow.weight),
+        order: conditionIndex++,
+        levelCombinationElements: factorialConditionTableRow.levels.map((level) => {
+          return {
+            id: uuidv4(),
+            level: level,
+          };
+        }),
+      });
+    });
+
+    return factorialConditionsRequestObject;
+  }
+
+  checkConditionTableValidity() {
+    const tableData = this.getFactorialConditionTableData();
+    let sumOfAssignmentWeights = 0.0;
+    let negativeweightError = false;
+
+    if (tableData.length > 0) {
+      tableData.forEach((factorialConditionTableRow) => {
+        sumOfAssignmentWeights += parseFloat(Number(factorialConditionTableRow.weight).toFixed(1));
+        parseFloat(Number(factorialConditionTableRow.weight).toFixed(1)) < 0 ? (negativeweightError = true) : null;
+      });
+
+      // checking if sum is not equal to 100
+      if (Math.round(sumOfAssignmentWeights) !== 100.0) {
+        return true || negativeweightError;
```

Comment:
this is will always return true, ` || negativeweightError` will never evaluate.

Reply:
done